### PR TITLE
Bi 14765 increase excel download

### PR DESCRIPTION
**Pull request by artemisgardnerch**
ensuring that the excel file downloaded contains 30,000 results

### CODE REVIEW

File: src/controllers/utils/utils.ts [hunk 1]
```diff
@@ -466,6 +466,7 @@ export const mapCompanySubtypeCheckboxes = (companySubtype: string | null | unde
 
 export const mapCompanyResource = (companyResource) => {
     const excludedCompanyTypeIncorpDates = ["registered-overseas-entity", "protected-cell-company"];
+    //this is the bottleneck
```
Comment (tombradley1) [inline src/controllers/utils/utils.ts]: What is it here that is causing the bottleneck? I think there is a node profiling tool that might help diagnose specifically the calls that are the slowest


Just a thought, but are there any gains to be had here by replacing the map with a for loop?
As a map will copy the values, will add some overhead, rather than a for loop that will accesses them using their index, but unsure if there's value at only 30k items?

File: src/model/page.urls.ts [hunk 2]
```diff
@@ -8,4 +8,4 @@ export const DISSOLVED_GET_RESULTS: string = "/dissolved-search/get-results";
 
 export const ADVANCED_ROOT: string = "/advanced-search";
 export const ADVANCED_GET_RESULTS: string = "/advanced-search/get-results";
-export const ADVANCED_DOWNLOAD: string = "/advanced-search/download";
+export const ADVANCED_DOWNLOAD: string = "/advanced-search/csv";
```
Comment (tombradley1) [inline src/model/page.urls.ts:11]: Will changing the url cause issues for higher regions/end users who will have this automated via a script etc? Would we need to notify them of this url change?

File: test/client/apiclient.test.ts [hunk 3]
```diff
@@ -246,4 +246,21 @@ describe("api.client", () => {
             chai.expect(advancedSearchResults).to.equal(mockAdvancedResponse.resource);
         });
     });
+
+    describe("advanced search", () => {
```
Comment (tombradley1) [inline test/client/apiclient.test.ts:250]: Your code checks for both 201 and 200, would be good to have a test case also for the 201 response

File: test/client/apiclient.test.ts [hunk 4]
```diff
@@ -246,4 +246,21 @@ describe("api.client", () => {
...
+            };
+            sandbox.stub(AdvancedSearchService.prototype, "getCompaniesAsCsv")
+                .returns(Promise.resolve(mockCsvResponse));
+
+            const searchParams: AdvancedSearchParams = createDummyAdvancedSearchParams("1", "testCompanyNameIncludes", "testCompanyNameExcludes", "testLocation", "01/01/2000",
+                "01/01/2001", "07210", "active", "ltd", "community-interest-company", "01", "10", "2010", "01", "03", "2010", 20);
+
+            const advancedSearchResults = await getAdvancedCompaniesAsCsv("api key", searchParams, "request id");
+            chai.expect(advancedSearchResults).to.equal(mockCsvResponse.resource);
+        });
```
Comment (tombradley1) [inline test/client/apiclient.test.ts:303]: Does there need to be a test for the negative path?
Perhaps asserting that if a non 201/200 response is given, that the correct params are given to the createError method?

**Review by tombradley1 — APPROVED** [review]
